spec/consensus/light-client/detection.md

```diff
@@ -1,3 +1,459 @@
+To obtain a new header, the light client first does [verification](verification)
+ with the primary, and then cross-checks the header
+with the secondaries using this specification.
+
+### Informal Problem statement
+
+> We put tags to informal problem statements as there is no sequential
+> secification.
+
+The following requirements are operational in that they describe how things
+should be done, rather than what should be done. They capture the intuition
+after the [light client ADR](lightclient) had been written. However, they do not constitute
+temporal logic verification conditions. For those, see [LCD-VC-*] below.
+
+#### **[LCD-IP-STATE]**
+
+We assume that `State` mentioned in [light client ADR](lightclient) is
+a collection of pairs *(fn,h)* that capture that header *h* was
+downloaded from full node *fn*.
+
+#### **[LCD-IP-Q]**
+
+Whenever the light client verifier adds a new pair
+*(p,h)* containing the primary *p* and a header *h* to *State*, the
+failure detector should query the secondaries by calling `Commit` remotely.
+
+
+#### **[LCD-IP-RespOK]**
+
+If a header *h'*, returned by the secondary *s*, is
+equal to *h* the failure detector adds *(s,h)* to state.
+
+
+*Remark:* This information might later be useful in case we find a
+problem when we get another header for this height from a different
+secondary. TODO: shall we keep that? Or is it sufficient to only store
+headers in `State`?
+
+
+#### **[LCD-IP-RespBad]**
+
+Otherwise, that is, if *h'* returned by *s* is
+different from *h*, the failure detector has to analyze the situation. If the failure detector
+can prove a fork on the main chain by performing the bisection protocol with *s*, it stops the
+light client and submits evidence.
+
+
+#### **[LCD-IP-PEERSET]**
+
+Whenever the failure detector observes misbehavior of a full node from
+the set of Secondaries it should be replaced by a fresh full node. (A full node
+that has not been primary or secondary before).
+
+
+
+## Assumptions/Incentives/Environment
+
+It is not in the interest of faulty full nodes to talk to the failure
+detector as long as the failure detector is connected to at least one
+correct full node. This would only increase the likelihood of
+misbehavior being detected. Also we cannot punish them easily
+(cheaply). The absence of a response need not be the fault of the full
+node. 
+
+Correct full nodes have the incentive to respond, because the failure
+detector may help them to understand whether their header is a good
+one. We can thus base liveness arguments of the failure detector on
+the assumptions that correct full nodes reliably talk to the failure
+detector.
+
+
+**Assumptions**
+
+#### **[LCD-A-CorrFull]**
+
+At all times there is at least one correct full
+node among the primary and the secondary.
+
+*Remark:* Perhaps [LCD-A-CorrFull] is not needed in the end because
+the verification conditions [LCD-VC-*] have preconditions on specific
+cases where primary and/or secondaries are faulty.
```

Comment:
When you say that `[LCD-VC-*]` have specific preconditions, I guess, it means that the conjunction of these preconditions is equivalent to `[LCD-A-CorrFull]` (it may be an implication).

Reply:
That is not so clear. The preconditions also talk about correct full nodes being on different branches. I don't believe that an abstract property as [LCD-A-CorrFull] be useful at all in describing good cases where we can detect something.